shiny/_namespaces.py

```diff
@@ -81,6 +81,10 @@ def resolve_id_or_none(id: Id | None) -> ResolvedId | None:
 
 
 def validate_id(id: str) -> None:
+    if not isinstance(id, str):
+        raise ValueError("`id` must be a single string")
+    if isinstance(id, str) and len(id) == 0:
```

Comment:
This can be simplified to:

```suggestion
    if id == "":
```